src/main/java/plugily/projects/villagedefense/events/Events.java

```diff
@@ -300,6 +301,16 @@ public void onSwap(CBPlayerSwapHandItemsEvent event) {
     }
   }
 
+  @EventHandler
+  public void onDecay(LeavesDecayEvent event) {
+    for (Arena arena : ArenaRegistry.getArenas()) {
+      if (!event.getBlock().getWorld().equals(arena.getStartLocation().getWorld())) {
+        continue;
+      }
+      event.setCancelled(true);
```

Comment:
Check if the `arena.getStartLocation()` not null and remove the complexity of this for loop by removing `continue`.
```suggestion
      Location startLoc = arena.getStartLocation();
      if (startLoc != null && event.getBlock().getWorld().equals(startLoc.getWorld())) {
        event.setCancelled(true);
      }
```
Also, I think we can break this for loop from processing because the event is cancelled?